contracts/src/core/BatcherPayments.sol

```diff
@@ -0,0 +1,136 @@
+pragma solidity =0.8.12;
+
+import {Initializable} from "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import {OwnableUpgradeable} from "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin-upgrades/contracts/security/PausableUpgradeable.sol";
+
+contract BatcherPayments is Initializable, OwnableUpgradeable, PausableUpgradeable {
+
+    // EVENTS
+    event PaymentReceived(address indexed sender, uint256 amount);
+    event FundsWithdrawn(address indexed recipient, uint256 amount);
+    event CreatedNewTask(
+        bytes32 batchMerkleRoot,
+        string batchDataPointer,
+        address[] proofSubmitters,
+        uint256 costOfRespondToTask
+    );
```

Comment:
I think this event is not needed, and it may add innecesary costs to the createNewTask transaction since the `proofSubmitters` array could grow significantly